crates/block-producer/src/mempool/tests.rs

```diff
@@ -16,6 +16,55 @@ impl Mempool {
     }
 }
 
+// OTEL TRACE TESTS
+// ================================================================================================
+
+#[tokio::test]
+async fn add_transaction_traces_are_correct() {
+    let (mut rx_export, _rx_shutdown) = miden_node_utils::logging::setup_test_tracing().unwrap();
+
+    let mut uut = Mempool::for_tests();
+    let txs = MockProvenTxBuilder::sequential();
+    uut.add_transaction(txs[0].clone()).unwrap();
+
+    let span_data = rx_export.recv().await.unwrap();
+    assert_eq!(span_data.name, "mempool.add_transaction");
+    assert!(span_data.attributes.iter().any(|kv| kv.key == "code.namespace".into()
+        && kv.value == "miden_node_block_producer::mempool".into()));
+    assert!(
+        span_data
+            .attributes
+            .iter()
+            .any(|kv| kv.key == "tx".into() && kv.value.to_string().starts_with("0x"))
+    );
+}
+
+#[tokio::test]
+async fn revert_transactions_traces_are_correct() {
+    let (mut rx_export, _rx_shutdown) = miden_node_utils::logging::setup_test_tracing().unwrap();
+
+    let mut uut = Mempool::for_tests();
+    let txs = MockProvenTxBuilder::sequential();
+    uut.add_transaction(txs[0].clone()).unwrap();
+    let span_data = rx_export.recv().await.unwrap();
+    assert_eq!(span_data.name, "mempool.add_transaction");
+
+    uut.revert_transactions(vec![txs[0].id()]).unwrap();
+    let span_data = rx_export.recv().await.unwrap();
+    assert_eq!(span_data.name, "mempool.revert_transactions");
+    assert!(span_data.attributes.iter().any(|kv| kv.key == "code.namespace".into()
+        && kv.value == "miden_node_block_producer::mempool".into()));
+
+    // TODO: Why do these fail? Am I misunderstanding something?
+    assert!(span_data.attributes.iter().any(|kv| kv.key == "transactions.ids".into()));
+    assert!(
+        span_data
+            .attributes
+            .iter()
+            .any(|kv| kv.key == "transactions.expired.ids".into())
+    );
```

Comment:
@Mirko-von-Leipzig I was expecting these to be here as per the above tests, but something must be different - do you know what it is?

```rust
    #[instrument(target = COMPONENT, name = "mempool.revert_transactions", skip_all, fields(transactions.ids))]
    fn revert_transactions(
        &mut self,
        txs: Vec<TransactionId>,
    ) -> Result<(), GraphError<TransactionId>> {
        tracing::Span::current().record("transactions.expired.ids", tracing::field::debug(&txs));
```

Reply:
Its because `tracing` doesn't let you dynamically add new properties. We've added `transactions.ids` as the field name, but are using a different one here.

Probably a copy-paste mistake/bug :)

Tests putting in some work already :partying_face:

Reply:
h u g e